test/healthcheck_test.go

```diff
@@ -342,6 +348,24 @@ func TestHealthCheckWithGoAway(t *testing.T) {
 
 func TestHealthCheckWithConnClose(t *testing.T) {
 	defer leakcheck.Check(t)
+
+	hcExitChan := make(chan struct{})
+	testHealthCheckFuncWrapper := func(ctx context.Context, newStream func() (interface{}, error), update func(bool), service string) error {
+		err := testHealthCheckFunc(ctx, newStream, update, service)
+		select {
```

Comment:
My context is long gone, but I remember feeling that this covered up the problem and didn't actually fix it.  Let's discuss offline.